App.js

```diff
@@ -0,0 +1,177 @@
+  let regex = /[-]?[.]?[0-9]*[.]?[0-9]+[-+/x][.]?[0-9]*[.]?[0-9]+/g;
+  let match = regex.exec(val);
+  if (match === null) {
+    return (val);
+  } else {
+    if (val.search('/') !== -1){
+      let aux1 = parseFloat(val.slice(0,val.search('/')));
+      let aux2 = parseFloat(val.slice(val.search('/') + 1));
+      if (
+        aux1 / aux2 === Infinity ||
+        aux1 / aux2 === -Infinity ||
+        isNaN(aux1 / aux2)
+      ) {
+        return ('mathError');
+      }
+      return (`${aux1 / aux2}`);
+    }
+    if (val.search('x') !== -1){
+      let aux7 = val.slice(0,val.search('x'));
+      let aux8 = val.slice(val.search('x') + 1);
+      return (`${aux7 * aux8}`);
+    }
+    if (val.search(/\+/) !== -1){
+      let aux3 = parseFloat(val.slice(0,val.search(/\+/)));
+      let aux4 = parseFloat(val.slice(val.search(/\+/) + 1));
+      return (`${aux3 + aux4}`);
+    }
+    if (val.search('-') !== -1){
+      let auxx = 1;
+      if (val[0] === '-') {
+        val = val.slice(1);
+        auxx = -1;
+      }
+      let aux5 = auxx * val.slice(0,val.search('-'));
+      let aux6 = val.slice(val.search('-') + 1);
+      return (`${aux5 - aux6}`);
+    }
+  }
+
+};
+
+const pointButt = val => {
+  if (/[a]+/g.test(val)){
+    return '.';
+  }
+  let myRe = /[^-+/x]+$/g;
+  let match = myRe.exec(val);
+  if (match === null) {
+    return (val + '.');
+  } else if (!match[0].includes('.')){
+    return (val + '.');
+  } else {return val;}
+};
+
+const calcOperator = (val, char) => {
+  if (/[a]+/g.test(val)){
+    return '';
+  }
+  if (val.length === 0 && char === '-'){
+    return (char);
+  } else if (val.length === 0){return val;}
+  else {
+    if (val.length === 1 && val[0] === '-'){
+      return val;
+    }
+    let character = val.charAt(val.length - 1);
+    if (
+      character === '-' ||
+      character === '+' ||
+      character === 'x' ||
+      character === '/'
+    ) {
+      return (val.slice(0, -1) + char);
+    } else if (character !== '.') {
+        if (
+          val.includes('-') ||
+          val.includes('/') ||
+          val.includes('x') ||
+          val.includes('+')
+        ) {
+          return (solveEquation(val) + char);
+        } else {
+          return (val + char);
+        }
+    } else {
+      return val;
+    }
+  }
+};
+
+const delOperator = (val) => {
+  if (/[a]+/g.test(val)){
+    return '';
+  }
+  if (val.length !== 0) {
+    return (val.slice(0, -1));
+  }
+  else {return val;}
+};
+
+const numButton = (val,num) => {
+  if (/[a]+/g.test(val)){
+    return num;
+  }
+  return (val + num);
+};
+
+const getScreenButtons = setVal => [
+  {label: '7', onPress: () => {setVal(prevValue => numButton(prevValue, '7'));}},
+  {label: '8', onPress: () => {setVal(prevValue => numButton(prevValue, '8'));}},
+  {label: '9', onPress: () => {setVal(prevValue => numButton(prevValue, '9'));}},
+  {label: '4', onPress: () => {setVal(prevValue => numButton(prevValue, '4'));}},
```

Comment:
Here we we are repeating a lot of code. We could could do as follow:

```suggestion
const getScreenButtons = setVal => [
  {label: '7', onPress: () => {setVal(prevValue => numButton(prevValue, '7'));}},
  {label: '8', onPress: () => {setVal(prevValue => numButton(prevValue, '8'));}},
  {label: '9', onPress: () => {setVal(prevValue => numButton(prevValue, '9'));}},
  {label: '4', onPress: () => {setVal(prevValue => numButton(prevValue, '4'));}},
```
```suggestion
const getScreenButtons = (value, setVal, operation) => [
  {label: value, onPress: () => {setVal(prevValue => operation(prevValue, value));}},
```